### PR TITLE
VP9e - use C++ primitives (memcpy => std::copy)

### PR DESCRIPTION
**Pull request by ilapichev**
Issue: MDP-50976

### CODE REVIEW

File: _studio/mfx_lib/encode_hw/vp9/include/mfx_vp9_encode_hw_utils.h [hunk 1]
```diff
@@ -274,7 +274,7 @@ enum // identifies memory type at encoder input w/o any details
     template<class T, class U> inline void Copy(T & dst, U const & src)
     {
         static_assert(sizeof(T) == sizeof(U), "copy objects of different size");
-        memcpy_s(&dst, sizeof(dst), &src, sizeof(dst));
+        dst = src;
```
Comment (lakulako) [inline _studio/mfx_lib/encode_hw/vp9/include/mfx_vp9_encode_hw_utils.h]: Types can be nonconvertible or implicit conversion may happen. It is not the same as memcpy. Would be better to delete this function.

File: _studio/mfx_lib/encode_hw/vp9/include/mfx_vp9_encode_hw_utils.h [hunk 2]
```diff
@@ -965,11 +965,11 @@ inline void CopySegmentationBuffer(mfxExtVP9Segmentation & dst, mfxExtVP9Segment
 {
     mfxU8* tmp = dst.SegmentId;
     ZeroExtBuffer(dst);
-    memcpy_s(&dst, sizeof(mfxExtVP9Segmentation), &src, sizeof(mfxExtVP9Segmentation));
+    dst = src;
     dst.SegmentId = tmp;
     if (dst.SegmentId && src.SegmentId && dst.NumSegmentIdAlloc)
     {
-        memcpy_s(dst.SegmentId, dst.NumSegmentIdAlloc, src.SegmentId, dst.NumSegmentIdAlloc);
+        dst = src;
```
Comment (lakulako) [inline _studio/mfx_lib/encode_hw/vp9/include/mfx_vp9_encode_hw_utils.h]: dst.SegmentId == src.SegmentId would be enough. Some space for optimization.

File: _studio/mfx_lib/encode_hw/vp9/src/mfx_vp9_encode_hw.cpp [hunk 3]
```diff
@@ -918,7 +918,8 @@ mfxStatus MFXVideoENCODEVP9_HW::GetVideoParam(mfxVideoParam *par)
 inline void UpdatePictureHeader(mfxU32 frameLen, mfxU32 frameNum, mfxU8* pPictureHeader, mfxU32 bufferSize)
 {
     mfxU32 ivf_frame_header[3] = {frameLen, frameNum << 1, 0x00000000};
-    memcpy_s(pPictureHeader, bufferSize, ivf_frame_header, sizeof (ivf_frame_header));
+    mfxU32 frameHeaderSize = sizeof (ivf_frame_header);
+    std::copy(ivf_frame_header, ivf_frame_header + std::min(bufferSize, frameHeaderSize), pPictureHeader);
```
[the remark this answers is not in the document]
Comment (lakulako) [reply _studio/mfx_lib/encode_hw/vp9/src/mfx_vp9_encode_hw.cpp]: Is it possible or makes sense bufferSize < frameHeaderSize?

File: _studio/mfx_lib/encode_hw/vp9/include/mfx_vp9_encode_hw_utils.h (hunk 1 above)
Comment (mgonchar) [reply _studio/mfx_lib/encode_hw/vp9/include/mfx_vp9_encode_hw_utils.h]: I agree, suggest to fully remove this function and replace with operator =

**Comment by dvrogozh** [comment]
@ilapichev: you are based on outdated forked copy, that's why tests are failing. Please, rebase your patch!

**Pull request closed without merging**